plugin-tray/lxqttray.cpp

```diff
@@ -373,11 +375,19 @@ void LXQtTray::stopTray()
 /************************************************
 
  ************************************************/
+void LXQtTray::onIconDestroyed(QObject * icon)
+{
+    mIcons.removeAll(qobject_cast<TrayIcon*>(icon));
```

Comment:
I can't test right now....but I think this could be the problem. The destroyed signal is emitted in QObject destructor and in that time the qobject_cast here returns nullptr. Can you @RalfJung test with static_cast/reinterpret_cast?